### PR TITLE
Array binary search

### PR DESCRIPTION
**Pull request by herndont**
Array Binary Search
- [x] Create a method called BinarySearch that  takes in 2 parameters: a sorted array and the search key. Without utilizing any of the built-in methods available to your language, return the index of the array’s element that is equal to the search key, or -1 if the element does not exist.
- [x] Utilize the Single-responsibility principle: any methods you write should be clean, reusable, abstract component parts to the whole challenge. You will be given feedback and marked down if you attempt to define a large, complex algorithm in one function definition.
- [x] For each method that you define, write test assertions for the following conditions at minimum: Happy Path, Expected Failure, and Edge Case
- [x] Stretch Goal: Once you’ve achieved a working solution, benchmark with varying array sizes (100, 10000, 100000 elements)

### CODE REVIEW

File: challengeReadme/BinarySearch.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+I also went ahead and tried to get some of the tests done and the stretch goals within the tests as well. I think that 
+I completed these correctly.
+
+
+The efficiency of this method would be log(n).
+
+### Solution
+####(White Board Solution)
+![image](../assets/array_binary_search.jpg)
```
Comment (JessLovell) [inline challengeReadme/BinarySearch.md:46]: this file path works locally, but does not work on github. Try using one '.' instead of two.  (./assets/array_binary_search.jpg)
